extension/lib/logger.js

```diff
@@ -108,24 +103,34 @@ Logger.prototype._writeAsync = function Logger_writeAsync(aMessage, aCallback) {
   // For testing purposes send the message to stdout
   //dump(aMessage + '\n');
 
-  // Create an output stream to write to file
-  var foStream = Cc["@mozilla.org/network/file-output-stream;1"]
-                 .createInstance(Ci.nsIFileOutputStream);
-  foStream.init(this._file, 0x02 | 0x08 | 0x10, PERMS_FILE, foStream.DEFER_OPEN);
-
-  // Write asynchronously to buffer;
-  // Input and output streams are closed after write
-  var iStream = this._converter.convertToInputStream(aMessage);
-  NetUtil.asyncCopy(iStream, foStream, function (status) {
-    if (!components.isSuccessCode(status)) {
-      var errorMessage = new Error("Error while writing to file: " + status);
-      console.error(errorMessage);
+  function callback(aResult) {
+    if (typeof(aCallback) === "function") {
+      aCallback(aResult);
     }
+  }
 
-    if (typeof(aCallback) === "function") {
-      aCallback(status);
+  let array = this._encoder.encode(aMessage);
+  let promise = OS.File.open(this.file.path, {write: true, append: true})
+  .then(file => {
+    file.write(array)
+    .then(bytes => {
+      file.close();
+      callback(bytes);
+    },
+    error => {
+      file.close();
+      callback(error);
+    })
+  })
+  .catch(error => {
+    // Extract something meaningful from WorkerErrorEvent
+    if (typeof error == "object" && error && error.constructor.name == "WorkerErrorEvent") {
+      let message = error.message;
+      throw new Error(message, error.filename, error.lineno);
```

Comment:
>     I don't see what specifically this code is doing. The
>     WorkerErrorEvent already has a message, and you now re-throw the
>     same error again?
> 
> All this |WorkerErrorEvent| is a workaround near Gecko 27 ... it should
> work on ESR31, but I'll test it.

We are not going to support anything below ESR31. Even that ESR release
is coming to its end of life in about 12 weeks. So remove any necessary
workaround for versions below ESR31. Thanks